### ISSUE TITLE
has_add_permission() takes 2 positional arguments but 3 were given

### ISSUE THREAD

**Issue by leunis** [celiao/django-rest-authemail#32]
I am getting this error when I click on a user in the admin.

'TypeError at /admin/core/user/1/change/
has_add_permission() takes 2 positional arguments but 3 were given'

I am a newbee in programming and in Django but I found on the internet that is has to to with inlines. So I have overwritten inlines in my custom EmailUserAdmin like so
```
class MyUserAdmin(EmailUserAdmin):
    fieldsets = (
        (None, {'fields': ('email', 'password')}),
        ('Personal Info', {'fields': ('first_name',
                                      'last_name',
                                      'is_company')}),
        ('Permissions', {'fields': ('is_active', 'is_staff',
                                    'is_superuser', 'is_verified')}),
        ('Important dates', {'fields': ('last_login', 'date_joined')}),
    )
    inlines = []
```
Of course I am missing some features now but the error is gone. How should I Solve this properly. This are my packages:

```
Django>= 3.1.1
djangorestframework>=3.9.0
psycopg2-binary
dj-database-url
gunicorn
django-cors-headers>=3.7.0
flake8>=3.6.0,<3.7.0
postmarker
django-rest-authemail
```

**Comment by leunis** [comment]
Hi,
I changed back to django 2.2.18. No problem here. So I guess something with has_permissions has changed in django 3.x

**Comment by DeeeeLAN** [comment]
Django 3.x modified django/contrib/admin/options.py at line 603: 
```
if not inline.has_add_permission(request, obj):
```
Updating all the authemail admin.py `has_add_permissions` from:
```
def has_add_permission(self, request):
    return False
```
to:
```
def has_add_permission(self, request, obj=None):
    return False
```
Fixes it for me on 3.x.

**Comment by pesfahanian** [comment]
> Django 3.x modified django/contrib/admin/options.py at line 603:
> 
> ```
> if not inline.has_add_permission(request, obj):
> ```
> 
> Updating all the authemail admin.py `has_add_permissions` from:
> 
> ```
> def has_add_permission(self, request):
>     return False
> ```
> 
> to:
> 
> ```
> def has_add_permission(self, request, obj=None):
>     return False
> ```
> 
> Fixes it for me on 3.x.

Thanks!
This worked fine.

**Comment by methodman75** [comment]
Also worked for me. Is there any way a fix can be made on the library side so that it works on both Django 2.2+ and 3+?

Perhaps by obtaining the django version 

```
import django
django.get_version()
```

Checking whether it is Django 3+

```
from distutils.version import StrictVersion
StrictVersion(django.get_version()) >= StrictVersion('3.0.0')
```

and modifying the `has_add_permission` signature using something like https://pypi.org/project/python-forge/ ?

```
import forge
if StrictVersion(django.get_version()) >= StrictVersion('3.0.0'):
  self.has_add_permission = forge.sign(
      forge.arg('self'),
      forge.arg('request'),
      forge.arg('obj', default=None),
  )(self.has_add_permission)
```

**Comment by DeeeeLAN** [comment]
I am pretty sure that by giving it a default (`obj=None`), it will work fine with either version.

**Comment by svandegar** [comment]
Thanks @DeeeeLAN, it worked fine for me. 

@celiao , any plan on merging the PR #41 ?

**Comment by kaepyro** [comment]
Thanks, it worked me too!
Please merge PR #41 .

**Comment by svandegar** [comment]
@celiao , would you mind having other mainteners joining the project, if you don't have time to review/merge PR's? 

I'm happy to help!

**Comment by arielaco** [comment]
Worked for me too. Bumping this.

**Comment by celiao** [comment]
Closed by #41
